tests/integration/components/new-signup/input-test.js

```diff
@@ -0,0 +1,132 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+// handler functions are creating some problems need to fix em
+module('Integration | Component | new-sign-up/form', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it has a label - What is your first name? when current step is firstStep', async function (assert) {
+    assert.expect(1);
+
+    this.setProperties({
+      onClick: function () {
+        this.currentStep = this.THIRD_STEP;
+      },
+      currentStep: 'firstName',
+    });
+
+    await render(hbs`
+      <NewSignup::Input
+        @onClick={{this.onClick}} 
+        @currentStep={{this.currentStep}}
+      />`);
+
+    assert
+      .dom('[data-test-signup-form-label]')
+      .hasText('What is your first name?');
+  });
+
+  test('it has a label - And what is your last name? when current step is lastName', async function (assert) {
```

Comment:
I can see that we are only asserting the labels here in tests. But our goal should be to test the feature. 

for example. If we have an email input field and it validates that the user should put an email in that field. we can test that our UI has a warning message that the email field has something else than email. 

Such type of tests gives us or confidence while shipping it to prod. and that the reason why we add tests.

Reply:
In our new signup flow, we have to enter one piece of information and then click `next` to be able to enter other information. so our tests should test that upon entering the first name and clicking the `next `button we are moved to the last name input field page. and so on.

Reply:
labels are basically the title of that particular input field, we are checking that if the label has changed that means we are on that next step

Reply:
oh, I see still I feel like it can be better.